include/highfive/H5DataType.hpp

```diff
@@ -37,6 +38,16 @@ enum class DataTypeClass {
     Invalid
 };
 
+inline DataTypeClass operator|(DataTypeClass lhs, DataTypeClass rhs) {
+    using T = std::underlying_type<DataTypeClass>::type;
+    return static_cast<DataTypeClass>(static_cast<T>(lhs) | static_cast<T>(rhs));
+}
+
+inline DataTypeClass operator&(DataTypeClass lhs, DataTypeClass rhs) {
+    using T = std::underlying_type<DataTypeClass>::type;
+    return static_cast<DataTypeClass>(static_cast<T>(lhs) & static_cast<T>(rhs));
```

Comment:
This looks like bitfield manipulations to me, but when I look at `DataTypeClass` it's just an `enum class` without giving the individual values a particular value. In a small reproducer I get that they're numbered through: `0, 1, 2, 3, ..., n`. Therefore, I'm not sure I understand theses lines of code correctly.

Reply:
This is just some convenient trick to be able to check multiple instances of this enum with a specific value. Like this allows you to more elegantly write this

```c++
if ((dtype.getClass() & data_type.getClass()) == DataTypeClass::Float) {
    // ...
}
```

instead of this

```c++
if ((dtype.getClass() == DataTypeClass::Float) and (data_type.getClass() == DataTypeClass::Float)) {
    // ...
}
```

if it were enums instead of enum classes, they would implicitly convert to ints and you coudl do this without defining bitwise operators, but for enum classes, one needs to add the operator. Full disclosure, the trick comes (of course, where else) from here: https://softwareengineering.stackexchange.com/a/204566

Reply:
Let me rephrase: ~elegantly~ briefly
since not everyone will find that elegant.

Reply:
By assigning powers of two as values, the bit logic should workout.